utility/src/test/java/org/oppia/android/util/logging/ConsoleLoggerTest.kt

```diff
@@ -0,0 +1,163 @@
+package org.oppia.android.util.logging
+
+import android.app.Application
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.core.content.pm.ApplicationInfoBuilder
+import androidx.test.core.content.pm.PackageInfoBuilder
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.testing.junit.OppiaParameterizedTestRunner
+import org.oppia.android.testing.junit.OppiaParameterizedTestRunner.SelectRunnerPlatform
+import org.oppia.android.testing.junit.ParameterizedRobolectricTestRunner
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestCoroutineDispatchers
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.data.DataProvidersInjector
+import org.oppia.android.util.data.DataProvidersInjectorProvider
+import org.oppia.android.util.locale.testing.LocaleTestModule
+import org.robolectric.Shadows
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Suppress("FunctionName")
+@RunWith(OppiaParameterizedTestRunner::class)
+@SelectRunnerPlatform(ParameterizedRobolectricTestRunner::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = ConsoleLoggerTest.TestApplication::class
+)
+class ConsoleLoggerTest {
+  @Inject lateinit var context: Context
+  @Inject lateinit var consoleLogger: ConsoleLogger
+
+  @Inject
+  lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+
+  private val testVersionName = "1.0"
+  private val testVersionCode = 1
+
+  private val testTag = "tag"
+  private val testLogLevel: LogLevel = LogLevel.ERROR
+  private val testMessage = "test error message"
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    setUpApplicationForContext()
+  }
+
+  @Test
+  fun testConsoleLogger_logError_withMessage_logsMessage() = runBlockingTest {
+    val consoleJob = launch {
+      consoleLogger.logErrorMessagesFlow.collect {
+        assertThat(it.logTag).isEqualTo(testTag)
+        assertThat(it.logLevel).isEqualTo(testLogLevel.toString())
+        assertThat(it.fullErrorLog).isEqualTo(testMessage)
+      }
+    }
+
+    consoleLogger.e(testTag, testMessage)
+
+    testCoroutineDispatchers.advanceUntilIdle()
+
+    consoleJob.cancel()
+  }
+
```

Comment:
Can we verify that this event log has been added to the list of logs to be uploaded? (FakeEventLogger.)

Reply:
I will include a seperate tests for these in the AnalyticsController to make sure they have been added to the logs.